src/main/java/org/ow2/proactive/catalog/util/parser/WorkflowParser.java

```diff
@@ -258,7 +273,11 @@ private void handleElementWithMultipleValues(ImmutableList.Builder<KeyValueLabel
             }
         }, xmlStreamReader);
 
-        keyValueMapBuilder.add(new KeyValueLabelMetadataEntity(key[0], value[0], attributeLabel));
+        if (key[0] != null && value[0] != null) {
```

Comment:
I guess you could create utility functions to express better what all these conditions mean.